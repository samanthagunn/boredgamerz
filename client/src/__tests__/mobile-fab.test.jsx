import React, { useState as useStateMock } from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { useAuth0 } from "@auth0/auth0-react";
import FAB from "../components/mobile-fab";
import renderer from "react-test-renderer";
jest.mock("@auth0/auth0-react");
jest.mock("react", () => ({
    ...jest.requireActual("react"),
    useState: jest.fn(),
}));

const user = {
    name: "Christian Glassiognon",
    given_name: "Christian",
    picture:
        "https://lh3.googleusercontent.com/a-/AOh14GiG4KuyPH8-ToSqzoBLwPTvb-px4DboWJn_Gujs=s96-c",
};

const setState = jest.fn();
describe("The header component...", () => {
    describe("that is user authenticated...", () => {
        beforeAll(() => {
            useStateMock.mockImplementation((init) => [init, setState]);
            useAuth0.mockReturnValue({
                logout: jest.fn(),
                loginWithRedirect: jest.fn(),
                isAuthenticated: true,
                isLoading: false,
                user: user,
                });
        });

    describe("The FAB component...", () => {
        test("renders without crashing", () => {
            const { baseElement } = render(<FAB />);
            expect(baseElement).toBeDefined();
        });
        test("matches the snapshot.", () => {
            const tree = renderer.create(<FAB />).toJSON();
            expect(tree).toMatchSnapshot();
        });

    // describe("the mobile nav...", () => {
    //     test("icon renders", () =>{
    //         const {ge} 
    //     })

    // })



    });

});
});










// describe("The FAB component...", () => {
//     test("renders without crashing", () => {
//         const { baseElement } = render(<FAB />);
//         expect(baseElement).toBeDefined();
//     });
//     test("matches the snapshot.", () => {
//         const tree = renderer.create(<FAB />).toJSON();
//         expect(tree).toMatchSnapshot();
//     });
// });