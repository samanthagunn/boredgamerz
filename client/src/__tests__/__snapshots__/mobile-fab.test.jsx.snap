// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header component... that is user authenticated... The FAB component... matches the snapshot. 1`] = `
<ion-fab
  horizontal="end"
  slot="fixed"
  vertical="bottom"
>
  <ion-fab-button
    onClick={[Function]}
  >
    <ion-icon
      icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Arrow Up Circle</title><path d='M256 48C141.13 48 48 141.13 48 256s93.13 208 208 208 208-93.13 208-208S370.87 48 256 48zm91.36 212.65a16 16 0 01-22.63.09L272 208.42V342a16 16 0 01-32 0V208.42l-52.73 52.32A16 16 0 11164.73 238l80-79.39a16 16 0 0122.54 0l80 79.39a16 16 0 01.09 22.65z'/></svg>"
    />
  </ion-fab-button>
  <ion-fab-list
    side="top"
  >
    <ion-fab-button
      href="/profile"
      onClick={[Function]}
    >
      <ion-icon
        icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Person Circle</title><path d='M258.9 48C141.92 46.42 46.42 141.92 48 258.9c1.56 112.19 92.91 203.54 205.1 205.1 117 1.6 212.48-93.9 210.88-210.88C462.44 140.91 371.09 49.56 258.9 48zm126.42 327.25a4 4 0 01-6.14-.32 124.27 124.27 0 00-32.35-29.59C321.37 329 289.11 320 256 320s-65.37 9-90.83 25.34a124.24 124.24 0 00-32.35 29.58 4 4 0 01-6.14.32A175.32 175.32 0 0180 259c-1.63-97.31 78.22-178.76 175.57-179S432 158.81 432 256a175.32 175.32 0 01-46.68 119.25z'/><path d='M256 144c-19.72 0-37.55 7.39-50.22 20.82s-19 32-17.57 51.93C191.11 256 221.52 288 256 288s64.83-32 67.79-71.24c1.48-19.74-4.8-38.14-17.68-51.82C293.39 151.44 275.59 144 256 144z'/></svg>"
      />
    </ion-fab-button>
    <ion-fab-button
      href="/mygames"
      onClick={[Function]}
    >
      <ion-icon
        icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Dice</title><path stroke-linecap='round' stroke-linejoin='round' d='M448 341.37V170.61A32 32 0 00432.11 143l-152-88.46a47.94 47.94 0 00-48.24 0L79.89 143A32 32 0 0064 170.61v170.76A32 32 0 0079.89 369l152 88.46a48 48 0 0048.24 0l152-88.46A32 32 0 00448 341.37z' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M69 153.99l187 110 187-110M256 463.99v-200' class='ionicon-fill-none ionicon-stroke-width'/><ellipse cx='256' cy='152' rx='24' ry='16'/><ellipse cx='208' cy='296' rx='16' ry='24'/><ellipse cx='112' cy='328' rx='16' ry='24'/><ellipse cx='304' cy='296' rx='16' ry='24'/><ellipse cx='400' cy='240' rx='16' ry='24'/><ellipse cx='304' cy='384' rx='16' ry='24'/><ellipse cx='400' cy='328' rx='16' ry='24'/></svg>"
      />
    </ion-fab-button>
    <ion-fab-button
      href="/games"
      onClick={[Function]}
    >
      <ion-icon
        icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Location</title><circle cx='256' cy='192' r='32'/><path d='M256 32c-88.22 0-160 68.65-160 153 0 40.17 18.31 93.59 54.42 158.78 29 52.34 62.55 99.67 80 123.22a31.75 31.75 0 0051.22 0c17.42-23.55 51-70.88 80-123.22C397.69 278.61 416 225.19 416 185c0-84.35-71.78-153-160-153zm0 224a64 64 0 1164-64 64.07 64.07 0 01-64 64z'/></svg>"
      />
    </ion-fab-button>
    <ion-fab-button
      href="/create"
      onClick={[Function]}
    >
      <ion-icon
        icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Add Circle</title><path d='M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M256 176v160M336 256H176' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
      />
    </ion-fab-button>
  </ion-fab-list>
</ion-fab>
`;
